emcc.py

```diff
@@ -34,10 +34,15 @@
 CXX_ENDINGS = ('.cpp', '.cxx', '.cc', '.c++', '.CPP', '.CXX', '.CC', '.C++', '.ii')
 OBJC_ENDINGS = ('.m', '.mi')
 OBJCXX_ENDINGS = ('.mm', '.mii')
-SPECIAL_ENDINGLESS_FILENAMES = ('/dev/null',)
+# A map of platform-specific 'null device' file(s)
+# There are only two keys in the map: False for non-Windows (or for POSIX to be more precise) and True for Windows
+SPECIAL_ENDINGLESS_FILENAMES = {
+  False: ('/dev/null',),
+  True: ('nul', 'NUL')
+}
```

Comment:
Feels like indexing with a boolean like this is a bit non-idiomatic and down below with the `SPECIAL_ENDINGLESS_FILENAMES[WINDOWS]` as well. How about the code would look like

    SPECIAL_ENDINGLESS_FILENAMES = ['nul', 'NUL'] if WINDOWS else ['/dev/null']

then any other locations that use the value don't need to specifically index with `[WINDOWS]`, and it would look cleaner and not need a special comment to explain. Likewise, I'd prefer using a list here rather than a tuple, since a list feels like it makes more sense.

On Windows, I think we could as well support `/dev/null`, because it would then just work for MSYS/Cygwin, and I don't think anyone is looking to have a file `C:/dev/null` with meaningful content anyways. That might help some random build script emulate support on Windows, with probably no drawbacks. So we could have this even as

    SPECIAL_ENDINGLESS_FILENAMES = ['/dev/null']
    if WINDOWS: SPECIAL_ENDINGLESS_FILENAMES += ['nul', 'NUL']

Also, I think we could just rename `SPECIAL_ENDINGLESS_FILENAMES` to `FILENAMES_REPRESENTING_NULL_FILE` since there's only files representing null there (if we do later have some more in there, we could just rename or add some other name).

@kripken: would you agree with this?